From e2b7d6c8da9f67e126bcfb4f117c8a529e2eaa10 Mon Sep 17 00:00:00 2001
From: Jean-Yves Avenard <jyavenard@mythtv.org>
Date: Wed, 25 Jun 2014 16:12:30 +1000
Subject: [PATCH 03/16] Allocate VideoFrame buffer using av_malloc, and
 mythframe utility routines.

Will need to have memory aligned frame in a future commit
---
 mythtv/libs/libmythtv/videoout_xv.cpp |   14 ++++++++------
 1 file changed, 8 insertions(+), 6 deletions(-)

diff --git a/mythtv/libs/libmythtv/videoout_xv.cpp b/mythtv/libs/libmythtv/videoout_xv.cpp
index 82f388a..cb9da54 100644
--- a/mythtv/libs/libmythtv/videoout_xv.cpp
+++ b/mythtv/libs/libmythtv/videoout_xv.cpp
@@ -524,12 +524,13 @@ void VideoOutputXv::CreatePauseFrame(VOSType subtype)
 {
     if (av_pause_frame.buf)
     {
-        delete [] av_pause_frame.buf;
+        av_freep(&av_pause_frame.buf);
         av_pause_frame.buf = NULL;
     }
 
-    init(&av_pause_frame, FMT_YV12,
-         new unsigned char[vbuffers.GetScratchFrame()->size + 128],
+    unsigned char* buf =
+        (unsigned char*)av_malloc(vbuffers.GetScratchFrame()->size + 128);
+    init(&av_pause_frame, FMT_YV12, buf,
          vbuffers.GetScratchFrame()->width,
          vbuffers.GetScratchFrame()->height,
          vbuffers.GetScratchFrame()->size);
@@ -1277,7 +1278,7 @@ void VideoOutputXv::DeleteBuffers(VOSType subtype, bool delete_pause_frame)
     {
         if (av_pause_frame.buf)
         {
-            delete [] av_pause_frame.buf;
+            av_freep(&av_pause_frame.buf);
             av_pause_frame.buf = NULL;
         }
         if (av_pause_frame.qscale_table)
@@ -1436,7 +1437,8 @@ void VideoOutputXv::PrepareFrameMem(VideoFrame *buffer, FrameScanType /*scan*/)
 
     int out_width  = display_visible_rect.width()  & ~0x1;
     int out_height = display_visible_rect.height() & ~0x1;
-    unsigned char *sbuf = new unsigned char[out_width * out_height * 3 / 2];
+    int size       = buffersize(FMT_YV12, out_width, out_height);
+    unsigned char *sbuf = (unsigned char*)av_malloc(size);
     AVPicture image_in, image_out;
     static struct SwsContext  *scontext;
 
@@ -1482,7 +1484,7 @@ void VideoOutputXv::PrepareFrameMem(VideoFrame *buffer, FrameScanType /*scan*/)
         disp->Unlock();
     }
 
-    delete [] sbuf;
+    av_freep(&sbuf);
 }
 
 // this is documented in videooutbase.cpp
-- 
1.7.10.2

